### PR TITLE
Set up CI for repo using github actions

### PR DESCRIPTION
**Pull request by amisevsk**
### What does this PR do?
Add CI for pull requests that
1. Runs shellcheck on all scripts
2. Builds the registry
3. Builds an offline registry

To test, I added it to [my fork](https://github.com/amisevsk/che-devfile-registry/pull/1/checks)

### What issues does this PR fix or reference?
Fixes https://github.com/eclipse/che/issues/14867

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+name: CI
+
+on:
+  pull_request:
+    branches: 
+      - master
+
+jobs:
+  shellcheck:
+    runs-on: ubuntu-latest
```
Comment (nickboldt) [inline .github/workflows/main.yml]: could we do this on RHEL8 instead?

Comment (amisevsk) [reply .github/workflows/main.yml]: It's a GitHub thing, your options are mac, windows, or ubuntu :slightly_frowning_face: 

I could look into updating it to run in a rhel docker container, but this is my first go at configuring GitHub actions.

Comment (nickboldt) [reply .github/workflows/main.yml]: +1 for this PR, +2 if you want to do a followup.

**Comment by amisevsk** [comment]
What's left on this issue is adding branch protections in the settings tab on the devfile registry; I don't have access to this so someone else will have to do it.

**Comment by nickboldt** [comment]
Just checked: don't have perms to view https://github.com/eclipse/che-devfile-registry/settings either. Maybe @l0rd  or @benoitf  have permission?

**Pull request merged**